src/Azure.Functions.Cli/Kubernetes/KubernetesHelpers.cs

```diff
@@ -250,11 +275,68 @@ internal static string GetKedaResources(string @namespace)
                 }
             }
 
-            result = result.Concat(deployments).ToList();
+            SecretsV1 existingFuncKeysSecret = null;
+            SecretsV1 newKeysSecret = null;
+            if (httpFunctions.Any())
+            {
+                var funcKeys = FuncAppKeysHelper.CreateKeys(httpFunctions.Select(f => f.Key));
+                SecretsV1 keysSecret = null;
+                (string output, bool keysSecretExist) = await ResourceExists("secret", keysSecretCollectionName, @namespace, true);
+                if (keysSecretExist)
+                {
+                    existingFuncKeysSecret = TryParse<SecretsV1>(output);
+                    if (existingFuncKeysSecret?.Data?.Any() == true)
+                    {
+                        funcKeys = funcKeys.Where(item => !existingFuncKeysSecret.Data.ContainsKey(item.Key)).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        if (!funcKeys.Any())
+                        {
+                            keysSecret = existingFuncKeysSecret;
+                        }
+                    }
+                }
+
+                if (funcKeys.Any())
+                {
+                    newKeysSecret = GetSecret(keysSecretCollectionName, @namespace, funcKeys);
+                    keysSecret = newKeysSecret;
+                }
```

Comment:
This logic is incorrect. Try running the example in this comment https://github.com/Azure/azure-functions-core-tools/pull/1715#pullrequestreview-338871909 

And verify the end result in the cluster.